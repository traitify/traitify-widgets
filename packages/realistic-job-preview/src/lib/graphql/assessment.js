export const create = `
  mutation createAssessment(
    $localeKey: String!,
    $profileID: String!,
    $surveyID: String!
  ) {
    createAssessment(
      localeKey: $localeKey,
      profileId: $profileID,
      surveyId: $surveyID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const get = `
  query assessment(
    $assessmentID: ID!
  ) {
    assessment(
      id: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const path = "/realistic-job-previews/graphql";

export const start = `
  mutation startAssessment(
    $assessmentID: ID!
  ) {
    startAssessment(
      id: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const update = `
  mutation updateAssessmentAnswer(
    $answers: [AssessmentAnswer],
    $assessmentID: String!
  ) {
    updateAssessmentAnswer(
      answers: $answers,
      assessmentId: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;
