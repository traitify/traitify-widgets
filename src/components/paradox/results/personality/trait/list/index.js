import PropTypes from "prop-types";
import {useState} from "react";
import PersonalityTrait from "components/results/personality/trait/details";
import TraitifyPropTypes from "lib/helpers/prop-types";
import useDidMount from "lib/hooks/use-did-mount";
import useDidUpdate from "lib/hooks/use-did-update";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

function PersonalityTraitList({setElement, ...props}) {
  const {assessment, getOption, isReady, translate, ui} = props;
  const [showMore, setShowMore] = useState(false);
  const state = {};

  useDidMount(() => { ui.trigger("PersonalityTraits.initialized", {props, state}); });
  useDidUpdate(() => { ui.trigger("PersonalityTraits.updated", {props, state}); });

  const disabledComponents = getOption("disabledComponents") || [];
  if(disabledComponents.includes("PersonalityTraits")) { return null; }
  if(!isReady("results")) { return null; }

  const allowHeaders = getOption("allowHeaders");
  const text = translate(showMore ? "show_less" : "show_more");
  let traits = assessment.personality_traits;

  if(!showMore) { traits = traits.slice(0, 5); }

  return (
    <div className={style.container} ref={setElement}>
      {allowHeaders && (
        <>
          <div className={style.sectionHeading}>{translate("personality_traits")}</div>
          <div className={style.p}>{translate("personality_traits_description")}</div>
        </>
      )}
      {traits.map((trait) => (
        <PersonalityTrait key={trait.personality_trait.id} trait={trait} {...props} />
      ))}
      <div className={style.center}>
        <button className={style.toggle} onClick={() => setShowMore(!showMore)} type="button">{text}</button>
      </div>
    </div>
  );
}

PersonalityTraitList.defaultProps = {assessment: null};
PersonalityTraitList.propTypes = {
  assessment: PropTypes.shape({
    personality_traits: PropTypes.arrayOf(
      PropTypes.shape({
        personality_trait: PropTypes.shape({
          definition: PropTypes.string.isRequired,
          id: PropTypes.string.isRequired,
          name: PropTypes.string.isRequired,
          personality_type: PropTypes.shape({
            badge: PropTypes.shape({
              color_1: PropTypes.string.isRequired,
              image_medium: PropTypes.string.isRequired
            }),
            name: PropTypes.string.isRequired
          })
        }).isRequired
      }).isRequired
    )
  }),
  getOption: PropTypes.func.isRequired,
  isReady: PropTypes.func.isRequired,
  setElement: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};

export {PersonalityTraitList as Component};
export default withTraitify(PersonalityTraitList);
