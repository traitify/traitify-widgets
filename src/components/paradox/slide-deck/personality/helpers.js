export function completedSlides(slides) {
  return slides.filter(({likert_response: likertResponse, response}) => (
    likertResponse || response != null
  )).map(({id, likert_response: likertResponse, response, time_taken: timeTaken}) => ({
    id,
    likert_response: likertResponse,
    response,
    time_taken: timeTaken && timeTaken >= 0 ? timeTaken : 2
  }));
}

export function isFinished(slides) {
  return slides.length > 0 && slides.length === completedSlides(slides).length;
}

export function isFullscreen() {
  const fullscreen = document.fullscreenElement
    || document.webkitFullscreenElement
    || document.mozFullScreenElement
    || document.msFullscreenElement;

  return !!fullscreen;
}

export function loadingIndex(slides) {
  return slides.findIndex((slide) => (
    !slide.loaded && !slide.likert_response && slide.response == null
  ));
}

export function slideIndex(slides) {
  return slides.findIndex((slide) => (!slide.likert_response && slide.response == null));
}

export function toggleFullscreen(options) {
  const {current, element} = options;

  if(current) {
    const exitFullscreen = document.exitFullscreen
      || document.webkitExitFullscreen
      || document.mozCancelFullScreen
      || document.msExitFullscreen;

    exitFullscreen.apply(document);
  } else {
    const requestFullscreen = element.requestFullscreen
      || element.webkitRequestFullscreen
      || element.mozRequestFullScreen
      || element.msRequestFullscreen;

    requestFullscreen.apply(element);
  }
}

export function isReady(slides) {
  if(slides.length === 0) { return false; }

  const loadedIndex = loadingIndex(slides);
  const remainingSlidesLoaded = loadedIndex === -1;
  const nextSlidesLoaded = loadedIndex > slideIndex(slides) + 2;

  return remainingSlidesLoaded || nextSlidesLoaded;
}
