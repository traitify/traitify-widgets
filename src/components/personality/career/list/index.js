import useCareers from "lib/hooks/use-careers";
import useComponentEvents from "lib/hooks/use-component-events";
import useResults from "lib/hooks/use-results";
import useTranslate from "lib/hooks/use-translate";
import Career from "../details";
import HighlightedCareers from "./highlighted";
import style from "./style.scss";

export default function CareerList() {
  const {careers, fetching, getNextPage, moreCareers} = useCareers();
  const results = useResults();
  const translate = useTranslate();

  useComponentEvents("CareerList");

  if(!results) { return null; }

  return (
    <div className={style.container}>
      <HighlightedCareers />
      {careers.map(({career, score}) => (
        <Career key={career.id} career={{score, ...career}} />
      ))}
      <div className={style.center}>
        {fetching && <span>{translate("loading")}</span>}
        {!fetching && (
          <>
            {moreCareers && (
              <button key="more" className={style.more} onClick={getNextPage} type="button">{translate("show_more")}</button>
            )}
            {careers.length === 0 && <span key="none">{translate("no_careers")}</span>}
            {careers.length > 0 && !moreCareers && <span key="done">{translate("no_more_careers")}</span>}
          </>
        )}
      </div>
    </div>
  );
}
