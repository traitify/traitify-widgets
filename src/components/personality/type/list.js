import PersonalityTypeChart from "components/personality/type/chart";
import useComponentEvents from "lib/hooks/use-component-events";

export default function PersonalityTypeList() {
  useComponentEvents("PersonalityTypes");

  return <PersonalityTypeChart />;
}
