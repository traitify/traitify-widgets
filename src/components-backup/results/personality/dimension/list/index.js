import PropTypes from "prop-types";
import PersonalityDimensionChart from "components/results/personality/dimension/chart";
import PersonalityDimensionDetails from "components/results/personality/dimension/details";
import {sortByTypePosition} from "lib/helpers";
import TraitifyPropTypes from "lib/helpers/prop-types";
import useDidMount from "lib/hooks/use-did-mount";
import useDidUpdate from "lib/hooks/use-did-update";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

function PersonalityDimensionList({setElement, ...props}) {
  const {assessment, getOption, isReady, translate, ui} = props;
  const state = {};

  useDidMount(() => { ui.trigger("PersonalityDimensions.initialized", {props, state}); });
  useDidUpdate(() => { ui.trigger("PersonalityDimensions.updated", {props, state}); });

  if(!isReady("results")) { return null; }
  const disabledComponents = getOption("disabledComponents") || [];
  const disableChart = disabledComponents
    .some((c) => ["PersonalityDimensionChart", "PersonalityDimensionColumns"].includes(c));
  const disableDetails = disabledComponents.includes("PersonalityDimensionDetails");
  if(disableChart && disableDetails) { return null; }

  const allowHeaders = getOption("allowHeaders");
  const types = sortByTypePosition(assessment.personality_types);

  return (
    <div className={style.container} ref={setElement}>
      {allowHeaders && <div className={style.sectionHeading}>{translate("personality_breakdown")}</div>}
      {!disableChart && <PersonalityDimensionChart {...props} />}
      {!disableDetails && (
        <div>
          {types.map((type) => (
            <PersonalityDimensionDetails
              key={type.personality_type.id}
              type={type}
              {...props}
            />
          ))}
        </div>
      )}
    </div>
  );
}

PersonalityDimensionList.defaultProps = {assessment: null};
PersonalityDimensionList.propTypes = {
  assessment: PropTypes.shape({
    personality_types: PropTypes.arrayOf(
      PropTypes.shape({
        personality_type: PropTypes.shape({
          id: PropTypes.string.isRequired
        }).isRequired
      }).isRequired
    )
  }),
  getOption: PropTypes.func.isRequired,
  isReady: PropTypes.func.isRequired,
  setElement: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};

export {PersonalityDimensionList as Component};
export default withTraitify(PersonalityDimensionList);
