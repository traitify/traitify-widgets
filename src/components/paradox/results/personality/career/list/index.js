import PropTypes from "prop-types";
import {useEffect, useState} from "react";
import {careerOption} from "lib/helpers";
import usePrevious from "lib/hooks/use-previous";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import Career from "../details";
import style from "./style.scss";

function CareerResults(props) {
  const [careers, setCareers] = useState([]);
  const [fetching, setFetching] = useState(false);
  const [moreCareers, setMoreCareers] = useState(false);

  const {assessmentID, locale, isReady, traitify, translate, ui} = props;

  const prevProps = {
    assessmentID: usePrevious(assessmentID),
    locale: usePrevious(locale)
  };

  const abortExistingRequest = () => {
    if(ui.current["Careers.fetching"]) {
      const previousRequest = ui.current["Careers.request"];
      previousRequest && previousRequest.xhr && previousRequest.xhr.abort();
    }
  };

  const fetch = () => {
    const fetchParams = ui.current["Careers.fetch"];

    if(!fetchParams) { return; }

    const path = careerOption(props, "path") || `/assessments/${assessmentID}/matches/careers`;
    const params = {
      careers_per_page: careerOption(props, "perPage") || 20,
      locale_key: locale,
      paged: true,
      ...fetchParams
    };

    abortExistingRequest();
    ui.trigger("Careers.fetching", {props}, true);
    ui.trigger(
      "Careers.request",
      {props},
      traitify.get(path, params).then((response) => {
        const previousCareers = (params.page > 1) ? careers : [];

        setCareers(previousCareers.concat(response));
        setMoreCareers(response.length >= params.careers_per_page);
      })
    );
  };

  const updateFetching = () => {
    setFetching(ui.current["Careers.fetching"]);
  };

  const mergeParams = () => {
    ui.trigger("Careers.fetch", {props}, {
      ...ui.current["Careers.fetch"],
      ...ui.current["Careers.mergeParams"]
    });
  };

  const showMore = () => {
    if(ui.current["Careers.fetching"]) { return; }
    const params = ui.current["Careers.fetch"] || {};
    const page = (params.page || 1) + 1;

    ui.trigger("Careers.mergeParams", {props}, {page});
  };

  const updateParams = () => {
    ui.trigger("Careers.fetch", {props}, {
      ...ui.current["Careers.updateParams"]
    });
  };

  useEffect(() => {
    ui.trigger("CareerResults.initialized", {props});
    ui.on("Careers.fetch", fetch);
    ui.on("Careers.fetching", updateFetching);
    ui.on("Careers.mergeParams", mergeParams);
    ui.on("Careers.updateParams", updateParams);

    if(isReady("results") && !ui.current["Careers.fetch"]) {
      ui.trigger("Careers.fetch", {props}, {});
    }

    return () => {
      ui.off("Careers.fetch", fetch);
      ui.off("Careers.fetching", updateFetching);
      ui.off("Careers.mergeParams", mergeParams);
      ui.off("Careers.updateParams", updateParams);

      abortExistingRequest();
      ui.trigger("Careers.fetching", {props}, false);
      ui.trigger("Careers.fetch", {props}, null);
    };
  }, []);

  useEffect(() => {
    ui.trigger("CareerResults.updated", {props});

    const assessmentChanged = assessmentID !== prevProps.assessmentID;
    const assessmentReady = isReady("results");
    const existingRequest = ui.current["Careers.fetch"];
    const localeChanged = locale !== prevProps.locale;

    if(assessmentReady && !existingRequest) {
      ui.trigger("Careers.fetch", {props}, {});
    } else if(assessmentReady && (assessmentChanged || localeChanged)) {
      abortExistingRequest();
      ui.trigger("Careers.fetching", {props}, false);
      ui.trigger("Careers.fetch", {props}, assessmentChanged ? {} : existingRequest);
    } else if(assessmentChanged || localeChanged) {
      abortExistingRequest();
      ui.trigger("Careers.fetching", {props}, false);
      ui.trigger("Careers.fetch", {props}, null);
    }
  });

  useEffect(() => {
    ui.trigger("Careers.fetching", {props}, false);
  }, [careers]);

  return (
    isReady("results") && (
      <div className={style.container}>
        {careers.map((_career) => {
          const {career, score} = _career;

          return <Career key={career.id} career={{score, ...career}} {...props} />;
        })}
        <p className={style.center}>
          {fetching && <span>{translate("loading")}</span>}
          {!fetching && [
            moreCareers && (
              <button key="more" className={style.more} onClick={showMore} type="button">{translate("show_more")}</button>
            ),
            (careers.length === 0 && <span key="none">{translate("no_careers")}</span>),
            (careers.length > 0 && !moreCareers && <span key="done">{translate("no_more_careers")}</span>)
          ]}
        </p>
      </div>
    )
  );
}
CareerResults.propTypes = {
  assessmentID: PropTypes.string.isRequired,
  isReady: PropTypes.func.isRequired,
  locale: PropTypes.string.isRequired,
  options: PropTypes.shape({
    careerOptions: PropTypes.shape({
      perPage: PropTypes.number
    })
  }),
  traitify: TraitifyPropTypes.traitify.isRequired,
  translate: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};
CareerResults.defaultProps = {
  options: null
};
export {CareerResults as Component};
export default withTraitify(CareerResults);
