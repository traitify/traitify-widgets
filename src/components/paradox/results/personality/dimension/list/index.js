import PropTypes from "prop-types";
import PersonalityDimensionChart from "components/results/personality/dimension/chart";
import PersonalityDimensionDetails from "components/results/personality/dimension/details";
import {sortByTypePosition} from "lib/helpers";
import {useDidMount, useDidUpdate} from "lib/helpers/hooks";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

function PersonalityDimensionList({element, ...props}) {
  const {assessment, getOption, isReady, ui} = props;
  const state = {};

  useDidMount(() => { ui.trigger("PersonalityDimensions.initialized", {props, state}); });
  useDidUpdate(() => { ui.trigger("PersonalityDimensions.updated", {props, state}); });

  if(!isReady("results")) { return null; }
  const disabledComponents = getOption("disabledComponents") || [];
  const disableChart = disabledComponents
    .some((c) => ["PersonalityDimensionChart", "PersonalityDimensionColumns"].includes(c));
  const disableDetails = disabledComponents.includes("PersonalityDimensionDetails");
  if(disableChart && disableDetails) { return null; }

  const types = sortByTypePosition(assessment.personality_types);

  return (
    <div className={style.container} ref={element}>
      {!disableChart && <PersonalityDimensionChart {...props} />}
      {!disableDetails && (
        <div>
          {types.map((type) => (
            <PersonalityDimensionDetails
              key={type.personality_type.id}
              type={type}
              {...props}
            />
          ))}
        </div>
      )}
    </div>
  );
}

PersonalityDimensionList.defaultProps = {assessment: null, element: null};
PersonalityDimensionList.propTypes = {
  assessment: PropTypes.shape({
    personality_types: PropTypes.arrayOf(
      PropTypes.shape({
        personality_type: PropTypes.shape({
          id: PropTypes.string.isRequired
        }).isRequired
      }).isRequired
    )
  }),
  element: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({current: PropTypes.instanceOf(Element)})
  ]),
  getOption: PropTypes.func.isRequired,
  isReady: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};

export {PersonalityDimensionList as Component};
export default withTraitify(PersonalityDimensionList);
