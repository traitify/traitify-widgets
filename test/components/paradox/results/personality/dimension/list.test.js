import {Component} from "components/paradox/results/personality/dimension/list";
import ComponentHandler from "support/component-handler";
import {mockOptions} from "support/helpers";
import assessment from "support/json/assessment/dimension-based.json";

jest.mock("components/results/personality/dimension/details", () => (() => (<div className="mock">PersonalityDimensionDetails</div>)));
jest.mock("lib/with-traitify", () => ((value) => value));

describe("Paradox.PersonalityDimensionList", () => {
  let props;

  beforeEach(() => {
    props = {
      assessment,
      getOption: jest.fn().mockName("getOption"),
      isReady: jest.fn().mockName("isReady").mockReturnValue(true),
      setElement: jest.fn().mockName("setElement"),
      translate: jest.fn().mockName("translate").mockImplementation((value, options = {}) => `${value}, ${options}`),
      ui: {
        current: {},
        off: jest.fn().mockName("off"),
        on: jest.fn().mockName("on"),
        trigger: jest.fn().mockName("trigger")
      }
    };
  });

  describe("callbacks", () => {
    it("triggers initialization", () => {
      new ComponentHandler(<Component {...props} />);

      expect(props.ui.trigger).toHaveBeenCalledWith(
        "PersonalityDimensions.initialized",
        expect.objectContaining({
          props: expect.any(Object),
          state: expect.any(Object)
        })
      );
    });

    it("triggers update", () => {
      const component = new ComponentHandler(<Component {...props} />);
      props.ui.trigger.mockClear();
      component.updateProps();

      expect(props.ui.trigger).toHaveBeenCalledWith(
        "PersonalityDimensions.updated",
        expect.objectContaining({
          props: expect.any(Object),
          state: expect.any(Object)
        })
      );
    });
  });

  it("renders component", () => {
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders component with headers", () => {
    mockOptions(props.getOption, {allowHeaders: true});
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if disabled", () => {
    mockOptions(props.getOption, {disabledComponents: ["PersonalityDimensionDetails", "PersonalityDimensionChart"]});
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if results not ready", () => {
    props.assessment = null;
    props.isReady.mockImplementation((value) => value !== "results");
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders chart if details disabled", () => {
    mockOptions(props.getOption, {disabledComponents: ["PersonalityDimensionDetails"]});
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders details if chart disabled", () => {
    mockOptions(props.getOption, {disabledComponents: ["PersonalityDimensionChart"]});
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders details if columns disabled", () => {
    mockOptions(props.getOption, {disabledComponents: ["PersonalityDimensionColumns"]});
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });
});
