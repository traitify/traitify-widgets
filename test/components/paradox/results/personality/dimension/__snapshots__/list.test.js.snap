// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paradox.PersonalityDimensionList renders chart if details disabled 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <p>
      dimension_description, [object Object]
    </p>
    <ul
      className="columns"
    >
      <li
        className="container"
      >
        <div>
          <div
            className="icon"
            style={
              Object {
                "background": "rgba(41,183,112,0.085)",
                "border": "3px solid #29B770",
              }
            }
          >
            <img
              alt="Agreeableness badge, [object Object]"
              src="https://cdn.traitify.com/badges/96dbf653-6ea0-4dc0-94dc-f6db3247510c_medium"
            />
          </div>
          <h3>
            Agreeableness
          </h3>
          <h2>
            Medium
          </h2>
        </div>
        <div
          className="background high"
          style={
            Object {
              "background": "rgba(41,183,112,0.1)",
            }
          }
        />
      </li>
      <li
        className="container"
      >
        <div>
          <div
            className="icon"
            style={
              Object {
                "background": "rgba(255,204,59,0.085)",
                "border": "3px solid #FFCC3B",
              }
            }
          >
            <img
              alt="Extraversion badge, [object Object]"
              src="https://cdn.traitify.com/badges/bf796c03-a863-4f61-9a1a-1f4fe6e408e8_medium"
            />
          </div>
          <h3>
            Extraversion
          </h3>
          <h2>
            Medium
          </h2>
        </div>
        <div
          className="background medium"
          style={
            Object {
              "background": "rgba(255,204,59,0.1)",
            }
          }
        />
      </li>
      <li
        className="container"
      >
        <div>
          <div
            className="icon"
            style={
              Object {
                "background": "rgba(249,139,28,0.085)",
                "border": "3px solid #F98B1C",
              }
            }
          >
            <img
              alt="Conscientiousness badge, [object Object]"
              src="https://cdn.traitify.com/badges/0d29bdb5-b502-4036-a790-9479d588e8fb_medium"
            />
          </div>
          <h3>
            Conscientiousness
          </h3>
          <h2>
            Medium
          </h2>
        </div>
        <div
          className="background medium"
          style={
            Object {
              "background": "rgba(249,139,28,0.1)",
            }
          }
        />
      </li>
      <li
        className="container"
      >
        <div>
          <div
            className="icon"
            style={
              Object {
                "background": "rgba(66,176,219,0.085)",
                "border": "3px solid #42B0DB",
              }
            }
          >
            <img
              alt="Openness badge, [object Object]"
              src="https://cdn.traitify.com/badges/216e23f2-f27c-46ae-b4b1-00c97a01db22_medium"
            />
          </div>
          <h3>
            Openness
          </h3>
          <h2>
            Medium
          </h2>
        </div>
        <div
          className="background medium"
          style={
            Object {
              "background": "rgba(66,176,219,0.1)",
            }
          }
        />
      </li>
      <li
        className="container"
      >
        <div>
          <div
            className="icon"
            style={
              Object {
                "background": "rgba(239,97,94,0.085)",
                "border": "3px solid #ef615e",
              }
            }
          >
            <img
              alt="Emotional Stability badge, [object Object]"
              src="https://cdn.traitify.com/badges/16cd30cc-3d5c-4c7e-b13f-828853034334_medium"
            />
          </div>
          <h3>
            Emotional Stability
          </h3>
          <h2>
            Medium
          </h2>
        </div>
        <div
          className="background low"
          style={
            Object {
              "background": "rgba(239,97,94,0.1)",
            }
          }
        />
      </li>
    </ul>
    <div
      className="columnsBuffer"
    />
  </div>
</div>
`;

exports[`Paradox.PersonalityDimensionList renders component 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <p>
      dimension_description, [object Object]
    </p>
    <ul
      className="columns"
    >
      <li
        className="container"
      >
        <div>
          <div
            className="icon"
            style={
              Object {
                "background": "rgba(41,183,112,0.085)",
                "border": "3px solid #29B770",
              }
            }
          >
            <img
              alt="Agreeableness badge, [object Object]"
              src="https://cdn.traitify.com/badges/96dbf653-6ea0-4dc0-94dc-f6db3247510c_medium"
            />
          </div>
          <h3>
            Agreeableness
          </h3>
          <h2>
            Medium
          </h2>
        </div>
        <div
          className="background high"
          style={
            Object {
              "background": "rgba(41,183,112,0.1)",
            }
          }
        />
      </li>
      <li
        className="container"
      >
        <div>
          <div
            className="icon"
            style={
              Object {
                "background": "rgba(255,204,59,0.085)",
                "border": "3px solid #FFCC3B",
              }
            }
          >
            <img
              alt="Extraversion badge, [object Object]"
              src="https://cdn.traitify.com/badges/bf796c03-a863-4f61-9a1a-1f4fe6e408e8_medium"
            />
          </div>
          <h3>
            Extraversion
          </h3>
          <h2>
            Medium
          </h2>
        </div>
        <div
          className="background medium"
          style={
            Object {
              "background": "rgba(255,204,59,0.1)",
            }
          }
        />
      </li>
      <li
        className="container"
      >
        <div>
          <div
            className="icon"
            style={
              Object {
                "background": "rgba(249,139,28,0.085)",
                "border": "3px solid #F98B1C",
              }
            }
          >
            <img
              alt="Conscientiousness badge, [object Object]"
              src="https://cdn.traitify.com/badges/0d29bdb5-b502-4036-a790-9479d588e8fb_medium"
            />
          </div>
          <h3>
            Conscientiousness
          </h3>
          <h2>
            Medium
          </h2>
        </div>
        <div
          className="background medium"
          style={
            Object {
              "background": "rgba(249,139,28,0.1)",
            }
          }
        />
      </li>
      <li
        className="container"
      >
        <div>
          <div
            className="icon"
            style={
              Object {
                "background": "rgba(66,176,219,0.085)",
                "border": "3px solid #42B0DB",
              }
            }
          >
            <img
              alt="Openness badge, [object Object]"
              src="https://cdn.traitify.com/badges/216e23f2-f27c-46ae-b4b1-00c97a01db22_medium"
            />
          </div>
          <h3>
            Openness
          </h3>
          <h2>
            Medium
          </h2>
        </div>
        <div
          className="background medium"
          style={
            Object {
              "background": "rgba(66,176,219,0.1)",
            }
          }
        />
      </li>
      <li
        className="container"
      >
        <div>
          <div
            className="icon"
            style={
              Object {
                "background": "rgba(239,97,94,0.085)",
                "border": "3px solid #ef615e",
              }
            }
          >
            <img
              alt="Emotional Stability badge, [object Object]"
              src="https://cdn.traitify.com/badges/16cd30cc-3d5c-4c7e-b13f-828853034334_medium"
            />
          </div>
          <h3>
            Emotional Stability
          </h3>
          <h2>
            Medium
          </h2>
        </div>
        <div
          className="background low"
          style={
            Object {
              "background": "rgba(239,97,94,0.1)",
            }
          }
        />
      </li>
    </ul>
    <div
      className="columnsBuffer"
    />
  </div>
  <div>
    <div
      className="mock"
    >
      PersonalityDimensionDetails
    </div>
    <div
      className="mock"
    >
      PersonalityDimensionDetails
    </div>
    <div
      className="mock"
    >
      PersonalityDimensionDetails
    </div>
    <div
      className="mock"
    >
      PersonalityDimensionDetails
    </div>
    <div
      className="mock"
    >
      PersonalityDimensionDetails
    </div>
  </div>
</div>
`;

exports[`Paradox.PersonalityDimensionList renders details if chart disabled 1`] = `
<div
  className="container"
>
  <div>
    <div
      className="mock"
    >
      PersonalityDimensionDetails
    </div>
    <div
      className="mock"
    >
      PersonalityDimensionDetails
    </div>
    <div
      className="mock"
    >
      PersonalityDimensionDetails
    </div>
    <div
      className="mock"
    >
      PersonalityDimensionDetails
    </div>
    <div
      className="mock"
    >
      PersonalityDimensionDetails
    </div>
  </div>
</div>
`;

exports[`Paradox.PersonalityDimensionList renders details if columns disabled 1`] = `
<div
  className="container"
>
  <div>
    <div
      className="mock"
    >
      PersonalityDimensionDetails
    </div>
    <div
      className="mock"
    >
      PersonalityDimensionDetails
    </div>
    <div
      className="mock"
    >
      PersonalityDimensionDetails
    </div>
    <div
      className="mock"
    >
      PersonalityDimensionDetails
    </div>
    <div
      className="mock"
    >
      PersonalityDimensionDetails
    </div>
  </div>
</div>
`;

exports[`Paradox.PersonalityDimensionList renders nothing if disabled 1`] = `null`;

exports[`Paradox.PersonalityDimensionList renders nothing if results not ready 1`] = `null`;
